### DISCUSSION TITLE
brew install specific version of ack

### DISCUSSION THREAD

**Discussion by rrevi** [Homebrew/discussions#3251] [Everyday usage] [answered]
Hello,

In the following [Formula](https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/ack.rb), using the brew install command, how do I install a specific version? Where I would invoke the else statement in line 49 (meaning not to install HEAD but instead a specific version of ack).

Thanks in advance

**Comment by SMillerDev** [comment, marked as answer]
Homebrew does not allow specifying arbitrary versions to install. You could use `brew extract` to copy an older version of a formula but there is no telling if that works.

**Reply by rrevi** [reply]
Thanks @SMillerDev 

One small follow up question, so if brew doesn't allow/support specifying arbitrary versions to install, then what or how do you invoke the else statement in line 49 if the current version of the ack [Formula](https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/ack.rb)? It seems to me that it is there to support installs of versions other than HEAD, no?

**Reply by SMillerDev** [reply]
`brew install ack` will use that. `brew install --HEAD ack` will use the first part
